File: renu_customization/public/js/fetched_supplier_email_on_purchase_order.js
frappe.ui.form.on("Purchase Order", {
    supplier(frm) {
        if (!frm.doc.supplier) return;

        frappe.call({
            method: "frappe.client.get_list",
            args: {
                doctype: "Address",
                fields: ["name", "email_id"],
                filters: [
                    ["Dynamic Link", "link_doctype", "=", "Supplier"],
                    ["Dynamic Link", "link_name", "=", frm.doc.supplier]
                ],
                limit: 1
            },
            callback(r) {
                if (r.message && r.message.length && r.message[0].email_id) {
                    frm.set_value("add_email_on_purchase_order", r.message[0].email_id);
                } else {
                    frm.set_value("add_email_on_purchase_order", "");
                    frappe.msgprint("No email found in Supplier Address");
                }
            }
        });
    }
});

File: renu_customization/api/fetched_supplier_email_on_purchase_order.py
import frappe

def set_supplier_email(doc, method=None):
    """
    Fetch Supplier Address email and set it in
    add_email_on_purchase_order field
    """

    if not doc.supplier:
        doc.add_email_on_purchase_order = ""
        return

    addresses = frappe.get_all(
        "Address",
        filters=[
            ["Dynamic Link", "link_doctype", "=", "Supplier"],
            ["Dynamic Link", "link_name", "=", doc.supplier]
        ],
        fields=["email_id"],
        limit_page_length=1
    )

    if addresses and addresses[0].get("email_id"):
        doc.add_email_on_purchase_order = addresses[0]["email_id"]
    else:
        doc.add_email_on_purchase_order = ""
